packages/core/src/explorers/explorer.ts

```diff
@@ -0,0 +1,19 @@
+import { Chain } from "@starknet-react/chains";
+import { StarkscanExplorer } from "./starkscan";
+import { StarkcompassExplorer } from "./starkCompass";
+import { ViewblockExplorer } from "./viewblock";
+
+export interface Explorer {
+  // link to a block
+  block(hashOrNumber: string | number): string;
```

Comment:
I think this interface should be slightly different to allow for hex-encoded block numbers in the future if needed.

```suggestion
  block(hashOrNumber: { hash: string, number :  number }): string;
```

Reply:
Need to update the implementations to match.

Reply:
I think it adds more complexity while I check if the provided param is a hash or a number with `hashOrNumber.toString().startsWith("0x")`
This should work with hex-encoded block numbers as well

(added toString() for it to work with type number too)

Reply:
Another reason for my choice is also that it implies the user needs to give the hash or the number not both

Reply:
But I get your point, should I proceed with your way?

Reply:
Yes please :)